export/export.go

```diff
@@ -108,185 +108,239 @@ func NewExportService(client clientset.Interface, pods PodService, nodes NodeSer
 	}
 }
 
-// Get all resources from each service.
-func (s *Service) get(ctx context.Context) (*ResourcesForExport, error) {
+type ErrorGroupWithSemaphore struct {
+	g   *errgroup.Group
+	sem *semaphore.Weighted
+}
+
+func NewErrorGroupWithSemaphore(ctx context.Context) ErrorGroupWithSemaphore {
 	g, _ := errgroup.WithContext(ctx)
 	sem := semaphore.NewWeighted(int64(runtime.GOMAXPROCS(0)))
+	return ErrorGroupWithSemaphore{
+		g:   g,
+		sem: sem,
+	}
+}
+
+// Get all resources from each service.
+func (s *Service) get(ctx context.Context) (*ResourcesForExport, error) {
+	errgrp := NewErrorGroupWithSemaphore(ctx)
 	resources := ResourcesForExport{}
 
-	if err := sem.Acquire(ctx, 1); err != nil {
-		return nil, xerrors.Errorf("acquire semaphore: %w", err)
+	errgrp.listPods(ctx, &s.podService, &resources)
+	errgrp.listNodes(ctx, &s.nodeService, &resources)
+	errgrp.listPvs(ctx, &s.pvService, &resources)
+	errgrp.listPvcs(ctx, &s.pvcService, &resources)
+	errgrp.listStorageClasses(ctx, &s.storageClassService, &resources)
+	errgrp.listPcs(ctx, &s.priorityclassService, &resources)
+	errgrp.getSchedulerConfig(ctx, &s.schedulerService, &resources)
+
+	if err := errgrp.g.Wait(); err != nil {
+		return nil, xerrors.Errorf("get resources all: %w", err)
+	}
+	return &resources, nil
+}
+
+func (s *Service) Export(ctx context.Context) (*ResourcesForExport, error) {
+	resources, err := s.get(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("export resources all: %w", err)
+	}
+	return resources, nil
+}
+
+// Import all resources from posted data.
+// (1) Restart scheduler based on the data.
+// (2) Apply each resource to the scheduler.
+//     * If UID is not nil, an error will occur. (try to find existing resource by UID)
+// (3) Get all resources. (Separated the get function to unify the struct format.)
+func (s *Service) Import(ctx context.Context, resources *ResourcesForImport) error {
+	if err := s.schedulerService.RestartScheduler(resources.SchedulerConfig); err != nil {
+		return xerrors.Errorf("restart scheduler with imported configuration: %w", err)
+	}
+	errgrp := NewErrorGroupWithSemaphore(ctx)
+	errgrp.applyPcs(ctx, &s.priorityclassService, resources)
+	errgrp.applyStorageClasses(ctx, &s.storageClassService, resources)
+	errgrp.applyPvcs(ctx, &s.pvcService, resources)
+	errgrp.applyPvs(ctx, &s.pvService, &s.pvcService, resources)
```

Comment:
I noticed that there is a possibility to fail to establish a relationship with PersistentVolume and PersistentVolumeClaim when PersistentVolumeClaim hasn't been applied yet.

So, applyPvs() should wait for a finish of applyPvcs(). 
Would you give me some idea to implement another asynchronous processing on it.